ci/docker/datal.build.wanda.yaml

```diff
@@ -10,6 +10,6 @@ srcs:
   - python/requirements/ml/data-requirements.txt
   - python/requirements/ml/data-test-requirements.txt
 build_args:
-  - ARROW_VERSION=17.*
+  - ARROW_VERSION=18.*
```

Comment:
do you want to change the versions in maybe `data-requirements.txt` or something so that the `python/requirements_compiled` is updated too?

Reply:
hmm, we don't want to force users to install pyarrow 18 / the latest version of pyarrow. they should be able to choose any version within the general bounds (`pyarrow >= 6.0.1,<18`). so i think we shouldn't add any explicit pyarrow pin in `data-requirements.txt` or `data-test-requirements.txt`? cc @can-anyscale

Reply:
The `python/requirements_compiled.txt` indicates the versions we install in the rayproject/ray docker image (orthogonal to ray pypi). It's currently on 14 (https://github.com/ray-project/ray/blob/master/python/requirements_compiled.txt#L1595) so pretty outdated itself.

It's optional but I recommend to update it to the latest version yes (or whoever supports python 3.13 will eventually upgrade it anyway). If so, maybe do it in another PR; it's not a blocker in this PR.